src/libcollections/string.rs

```diff
@@ -1072,3 +1128,61 @@ impl fmt::Write for String {
         Ok(())
     }
 }
+
+/// A draining iterator for `String`.
+#[unstable(feature = "collections_drain", reason = "recently added")]
+pub struct Drain<'a> {
+    string: *mut String,
+    /// Start of part to remove
+    start: usize,
+    /// Index of tail to preserve
+    tail_start: usize,
+    /// Current remaining range to remove
+    iter: Chars<'a>,
+    _marker: PhantomData<&'a mut String>,
```

Comment:
Maybe it shouldn't be? AFAIU now is that &'a mut X and &'a X behave the same w.r.t the lifetime but not w.r.t X. So I think the PhantomData is just redundant, because the 'a variance is already correct.

Apart from variance, is there anything else that it does? Sync and Send are already handled manually.